core/src/mender-client.c

```diff
@@ -376,12 +340,6 @@ mender_client_register_artifact_type(char *type,
     mender_client_artifact_type_t **tmp;
     mender_err_t                    ret;
```

Comment:
Note to self,
```suggestion
    mender_err_t                    ret = MENDER_OK;
```

Here we are missing initialization. Please **review all of these**